src/typing_extensions.py

```diff
@@ -3007,6 +3136,12 @@ def wrapper(*args, **kwargs):
                 )
 
 
+def _is_param_expr(arg):
+    return arg is ... or isinstance(
+        arg, (tuple, list, ParamSpec, _ConcatenateGenericAlias)
```

Comment:
Do we need to cover the typing as well as the typing-extensions versions of the last two?

Reply:
For `_ConcatenateGenericAlias` we indeed need it. Fixed and added also tests for both.